backend/api/pretix/types.py

```diff
@@ -92,6 +163,86 @@ def type(self) -> TicketType:
 
         return TicketType.STANDARD
 
+    @classmethod
+    def from_data(
+        cls,
+        data: ItemDict,
+        language: str,
+        categories: Dict[str, CategoryDict],
+        questions: Dict[str, Question],
+        quotas: Optional[Dict[str, QuotaDict]] = None,
+    ):
+        category = _get_category_for_ticket(data, categories)
+        return cls(
+            id=data["id"],
+            language=language,
+            name=_get_by_language(data, "name", language),
+            description=_get_by_language(data, "description", language),
+            tax_rate=data["tax_rate"],
+            active=data["active"],
+            default_price=data["default_price"],
+            available_from=data["available_from"],
+            available_until=data["available_until"],
+            category=_get_by_language(category, "name", language),
+            category_internal_name=category.get("internal_name", None),
+            variations=[
+                ProductVariation.from_data(variation, language)
+                for variation in data.get("variations", [])
+            ],
+            questions=[
+                Question.from_data(question, language)
+                for question in questions
+                if data["id"] in question["items"]
+            ],
+            quantity_left=cls._get_quantity_left(data, quotas),
+        )
+
+    @staticmethod
```

Comment:
why a staticmethod?

Reply:
because I'm using it in `from_data` that is a `classmethod`, so I don't have the object instance... can be static or class method